Sources/Interaction/Manipulators/3DControllerModelSelectorManipulator/index.js

```diff
@@ -0,0 +1,204 @@
+    _interactorStyle,
+    renderer,
+    _state,
+    _device,
+    _input,
+    pressed,
+    targetPosition,
+    targetOrientation
+  ) => {
+    // If the button is not pressed, clear the state
+    if (!pressed) {
+      model.lastOrientation = null;
+      model.lastWorldPosition = null;
+      pickedProp = null;
+      return;
+    }
+
+    const camera = renderer.getActiveCamera();
+    const physicalToWorldMatrix = [];
+    camera.getPhysicalToWorldMatrix(physicalToWorldMatrix);
+
+    // Since the targetPosition we get is in physical coordinates,
+    // transform it using the physicalToWorldMatrix to get it in woorld coordinates
+    const targetRayWorldPosition = vec3.transformMat4(
+      [],
+      [targetPosition.x, targetPosition.y, targetPosition.z],
+      physicalToWorldMatrix
+    );
+
+    const targetRayWorldDirection = camera.physicalOrientationToWorldDirection([
+      targetOrientation.x,
+      targetOrientation.y,
+      targetOrientation.z,
+      targetOrientation.w,
+    ]);
+
+    const dist = renderer.getActiveCamera().getClippingRange()[1];
+    const rayPoint1 = [...targetRayWorldPosition, 1.0];
+    const rayPoint2 = [
+      rayPoint1[0] - targetRayWorldDirection[0] * dist,
+      rayPoint1[1] - targetRayWorldDirection[1] * dist,
+      rayPoint1[2] - targetRayWorldDirection[2] * dist,
+      1.0,
+    ];
+
+    // Perform picking on the given renderer
+    picker.pick3DPoint(rayPoint1, rayPoint2, renderer);
+    const props = picker.getActors();
+
+    // If we have picked props, store the first one.
+    if (props.length > 0) {
+      pickedProp = props[0];
+    }
+  };
+
+  publicAPI.onMove3D = (_interactorStyle, renderer, _state, eventData) => {
+    // If we are not interacting with any prop, we have nothing to do.
+    if (pickedProp == null) {
+      return;
+    }
+
+    const camera = renderer.getActiveCamera();
+    const physicalToWorldMatrix = [];
+    camera.getPhysicalToWorldMatrix(physicalToWorldMatrix);
+
+    const { targetPosition } = eventData;
+
+    const targetRayWorldPosition = vec3.transformMat4(
+      [],
+      [targetPosition.x, targetPosition.y, targetPosition.z],
+      physicalToWorldMatrix
+    );
+
+    // this is a unit quaternion
+    const targetRayOrientation = quat.fromValues(
+      eventData.targetOrientation.x,
+      eventData.targetOrientation.y,
+      eventData.targetOrientation.z,
+      eventData.targetOrientation.w
+    );
+
+    if (model.lastWorldPosition && model.lastOrientation) {
+      positionProp(pickedProp, targetRayWorldPosition, targetRayOrientation);
+    }
+
+    model.lastWorldPosition = [...targetRayWorldPosition];
+    model.lastOrientation = [...targetRayOrientation];
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  device: Device.RightController,
+  input: Input.TrackPad,
```

Comment:
why do you need to define them ? 
They probably have not been defined by mother class for a good reason.

Reply:
agree. let's be explicit.